packages/core/src/external/carequality/ihe-gateway-v2/inbound/xca/process/dr-request.ts

```diff
@@ -0,0 +1,48 @@
+import { XMLParser } from "fast-xml-parser";
+import { toArray } from "@metriport/shared";
+import { iti39RequestSchema, DocumentRequest } from "./schema";
+import { InboundDocumentRetrievalReq, DocumentReference } from "@metriport/ihe-gateway-sdk";
+import { convertSamlHeaderToAttributes, extractTimestamp } from "../../shared";
+import { stripUrnPrefix } from "../../../../../../util/urn";
+import { extractText } from "../../../utils";
+
+function extractDocumentReferences(documentRequest: DocumentRequest[]): DocumentReference[] {
+  return documentRequest.map(req => ({
+    homeCommunityId: stripUrnPrefix(req.HomeCommunityId),
+    docUniqueId: stripUrnPrefix(req.DocumentUniqueId),
+    repositoryUniqueId: stripUrnPrefix(req.RepositoryUniqueId),
+  }));
+}
+
+export function processInboundDrRequest(request: string): InboundDocumentRetrievalReq {
+  try {
+    const parser = new XMLParser({
+      ignoreAttributes: false,
+      attributeNamePrefix: "_",
+      textNodeName: "_text",
+      parseAttributeValue: false,
+      removeNSPrefix: true,
+    });
+    const jsonObj = parser.parse(request);
+    console.log("jsonObj", JSON.stringify(jsonObj, null, 2));
+    const iti39Request = iti39RequestSchema.parse(jsonObj);
+    const samlAttributes = convertSamlHeaderToAttributes(iti39Request.Envelope.Header);
+    const documentRequests = toArray(
+      iti39Request.Envelope.Body.RetrieveDocumentSetRequest.DocumentRequest
+    );
+    const documentReference = extractDocumentReferences(documentRequests);
+
+    return {
+      id: stripUrnPrefix(extractText(iti39Request.Envelope.Header.MessageID)),
+      timestamp: extractTimestamp(iti39Request.Envelope.Header),
+      samlAttributes,
+      documentReference,
+      signatureConfirmation: extractText(
+        iti39Request.Envelope.Header.Security.Signature.SignatureValue
+      ),
+    };
+  } catch (error) {
+    console.log(error);
```

Comment:
ditto: left-over? here and other `console.log` calls

generally speaking:
- to log, let's use `out.log`
- when logging, no new-lines
- log if needed (so logging on the handling code is useful if we're not logging upstream - which I assume we are and also capturing it)

Reply:
sorry about these. I just added them for when I was testing on the lambda just now

Reply:
No worries! I noticed there are some still left over. Please search for `console.` on this PR and you'll see:
- [a log that seems for debug on the lambda](https://github.com/metriport/metriport/pull/2223/files#diff-508814c9bba3f797030b850d8e9fc603e6dfbfde2a96fb0f581880cc5189ca06R57)
- [a multi-line log on core](https://github.com/metriport/metriport/pull/2223/files#diff-9d604ef2ec647d4b0fc86313f850a472b6dd539d1cac759b05516dba0eac41e9R76)